prime/src/pages/BorrowActionsPage.tsx

```diff
@@ -383,19 +382,22 @@ export default function BorrowActionsPage() {
     // tell React about our findings
     const _marginAccount = { ...marginAccount };
     if (userWantsHypothetical) {
-      const numericBorrowInterest = parseFloat(borrowInterestInputValue) || 0.0;
-      const numericSwapFees = parseFloat(swapFeesInputValue) || 0.0;
+      const token0Decimals = marginAccount.token0.decimals;
+      const token1Decimals = marginAccount.token1.decimals;
+      const selectedToken = isToken0Selected ? marginAccount.token0 : marginAccount.token1;
+      const numericBorrowInterest = GN.fromDecimalString(borrowInterestInputValue ?? '0', selectedToken.decimals);
+      const numericSwapFees = GN.fromDecimalString(swapFeesInputValue ?? '0', selectedToken.decimals);
       // Apply the user's inputted swap fees to the displayed margin account's assets
       _marginAccount.assets = {
         ...assetsF,
-        token0Raw: assetsF.token0Raw + (isToken0Selected ? numericSwapFees : 0),
-        token1Raw: assetsF.token1Raw + (isToken0Selected ? 0 : numericSwapFees),
+        token0Raw: assetsF.token0Raw.add(isToken0Selected ? numericSwapFees : GN.zero(token1Decimals)),
+        token1Raw: assetsF.token1Raw.add(isToken0Selected ? GN.zero(token0Decimals) : numericSwapFees),
```

Comment:
Decimals are flipped here. When adding to token0Raw, the addend should always have token0 decimals. When adding to token1Raw, the addend should always have token1 decimals.